Develop3

coldfredd opened the pull request.

coldfredd committed:
add allure report

coldfredd committed:
test

pomegranate282922 commented on src/main/java/diplom3/WebDriverFactory.java:
```diff
@@ -0,0 +1,19 @@
+            return  WebDriverManager.chromedriver().create();
+        }
+        if  (browserType.equals("yandex")) {
+            System.setProperty("webdriver.chrome.driver", "/Users/user/Downloads/yandexdriver");
```
Личных путей в коде быть не должно 
Нужно исправить

coldfredd replied:
исправил

pomegranate282922 commented on src/main/java/diplom3/page/LoginPage.java:
```diff
@@ -0,0 +1,59 @@
+        WebElement passwordInput = webDriver.findElement(passwordInputLoginLocator);
+        passwordInput.sendKeys(password);
+    }
+    public void clickRestorePasswordLink(){
```
Нужно расставить степы с наименованиями 
Нужно исправить

coldfredd replied:
исправил

pomegranate282922 commented on src/main/java/diplom3/page/MainPage.java:
```diff
@@ -0,0 +1,85 @@
+    private final By tabSaucesSelectedLocator = By.xpath("//div[contains(@class,'tab_tab_type')]/span[text()='Соусы']");//Таб Соусы Выбран
+    private final By SaucesLocator = By.xpath("//h2[text()='Соусы']");//Раздел Соусы
+    private final By tabFillingsLocator = By.xpath("//span[text()='Начинки']");//Таб Начинки
+    private final By tabFillingsSelectedLocator = By.xpath("//div[contains(@class,'tab_tab_type')]/span[text()='Начинки']");//Таб Начинки Выбран
```
Вообще ключевое-то current 
Можно улучшить

coldfredd replied:
Про название локатора имеете ввиду? Вместо tabSaucesSelectedLocator написать tabSaucesCurrentLocator?

coldfredd replied:
поменял названия локаторов

pomegranate282922 replied:
Не, про то какой класс мы ищем, точнее какое слово в нем.

pomegranate282922 commented on src/main/java/diplom3/steps/UserSteps.java:
```diff
@@ -0,0 +1,40 @@
+                .then();
+    }
+
+    public ValidatableResponse login(User user) {
```
Степы 
Нужно добавить

coldfredd replied:
добавил степы

pomegranate282922 commented on src/test/java/diplom3/AccountProfileTest.java:
```diff
@@ -0,0 +1,148 @@
+    @Before
+    public void setup(){
+        webDriver = WebDriverFactory.getWebDriver(BROWSER);
+        webDriver.get("https://stellarburgers.nomoreparties.site/");
```
Все Урлы выносим как константы 
Нужно исправить

coldfredd replied:
исправил

pomegranate282922 commented on src/test/java/diplom3/AccountProfileTest.java:
```diff
@@ -0,0 +1,148 @@
+    private void userLogin(LoginPage loginPage) {
+        loginPage.userLogin(user.getEmail(), user.getPassword());
+    }
+    @Step("Click Enter Button")
```
Зачем это дублирование? Расставь степы в po и убери эти методы отсюда. Они лишние же явно. 
нужно исправить

coldfredd replied:
убрал дублирование

pomegranate282922 reviewed (commented):
Работа хорошая! Отлично сделан выбор браузера. Есть рандомизация. Везде предусмотрено удаление данных. 
замечания:
1. Убрать дублирования степов 
2. Вынести Урлы 
3. Остальное по мелочи

coldfredd committed:
Исправил дублирования степов
Исправил ссылку на yandexdriver, теперь смотрит на папку resources
Вынес урл в константу
Поменял названия локаторов для выбранных табов на current

coldfredd committed:
убрал неиспользуемые импорты

pomegranate282922 reviewed (approved):
Вот и прекрасно!
